Supports more legalizations and multi-partition

neuropilot-captain opened the pull request:
1. more legalizations 
   a. Batch_MatMul
   b. Mul
   c. Fully_Connected

3. multi-partition supports 
    introduce utils for supporting multi-partition

neuropilot-captain committed:
Supports more legalizations and multi-partition

1. more legalizations:
  a. Batch_MatMul
  b. Mul
  c. Fully_Connected

2. multi-partition supports:
  introduce utils for supporting multi-partition

keerthanakadiri commented:
Hi @neuropilot-captain , Can you please sign and resolve the conflicts? Thank you!

neuropilot-captain committed:
Merge branch 'tensorflow:master' into upstream_0225

neuropilot-captain committed:
Support more legalizations for MTK platform

- Added new legalizatios for more operations.
- Introduced common utility functions for neuron.

YunanAZ commented on tensorflow/lite/experimental/litert/vendors/mediatek/schema/BUILD:
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2025 MediaTek Inc.
```
Please fix the syntax of BUILD.

neuropilot-captain replied:
I have tried to fix the syntax issues in the BUILD file. Please review the latest commit to ensure everything is correct. Thanks 😄 !

mihaimaruseac commented on tensorflow/lite/experimental/litert/vendors/mediatek/compiler/BUILD:
```diff
@@ -148,4 +163,4 @@ litert_test(
         "@com_google_absl//absl/strings:string_view",
     ],
-)
+)
```
Please terminate the file with a newline character (not an empty line though!)

neuropilot-captain replied:
Thank you for pointing that out. I have added a newline character at the end of the file as requested. If there are any other issues, please let me know😄.

neuropilot-captain committed:
Fix syntax error of the BUILD
